src/stores/room-list/RoomListStore.ts

```diff
@@ -496,7 +496,8 @@ export class RoomListStoreClass extends AsyncStoreWithClient<IState> implements
     private getPlausibleRooms(): Room[] {
         if (!this.matrixClient) return [];
 
-        let rooms = this.matrixClient.getVisibleRooms().filter((r) => VisibilityProvider.instance.isRoomVisible(r));
+        let rooms = this.matrixClient.getVisibleRooms(SettingsStore.getValue("feature_dynamic_room_predecessors"));
```

Comment:
We should avoid SettingsStore calls on the critical codepaths such as this as they end up being expensive

Reply:
Where should we call it?

Reply:
Class constructor and set up a watcher/monitor to respond to updates and recalculate the room list when such an update happens